include/nuttx/spinlock.h

```diff
@@ -353,6 +354,8 @@ spin_trylock_wo_note(FAR volatile spinlock_t *lock)
     }
 
   SP_DMB();
+  sched_lock_wo_note();
```

Comment:
lose sched lock note in spin_trylock

Reply:
spin_trylock will call spin_trylock_wo_note
spin_trylock_wo_note will call sched_lock_wo_note

Reply:
spin_trylock should call sched_lock to trace the preemption state.